File: elixir/acronym/acronym.exs
defmodule Acronym do
  @doc """
  Generate an acronym from a string.
  "This is a string" => "TIAS"
  """
  @spec abbreviate(String.t()) :: String.t()
  def abbreviate(string) do
    string
    |> String.split(~r/ |-|(?=[A-Z][^A-Z])/, trim: true)
    |> Enum.map(&prefix/1)
    |> Enum.join
  end

  def prefix(word) do
    word
    |> String.first
    |> String.upcase
  end
end

File: python/acronym/acronym.py
import re

def abbreviate(words):
  letters = []

  for word in words.replace('-', ' ').split():
    letters.append(_first_alpha(word))

  return ''.join(letters).upper()


def _first_alpha(word):
  for letter in word:
    if letter.isalpha():
      return letter
